File: java/yapay-gateway-client/src/main/java/br/com/yapay/gateway/builder/OneClickBuilder.java
package br.com.yapay.gateway.builder;

import br.com.yapay.gateway.model.Credential;
import br.com.yapay.gateway.model.OneClickRegisterData;

public class OneClickBuilder {

	private OneClickRegisterData oneClickData;

	private String storeCode;

	public OneClickBuilder() {

	}

	public OneClickBuilder(Credential credential) {
		this.storeCode = credential.getStoreCode();
	}

	public OneClickRegisterData build() {
		OneClickRegisterData newOneClickRegisterData = oneClickData;
		oneClickData = null;
		return newOneClickRegisterData;
	}

	public void newOneClickRegister(Long paymentCode) {
		this.newOneClickRegister(null, paymentCode);
		oneClickData.setStoreCode(storeCode);
	}

	public void newOneClickRegister(String storeCode, Long paymentCode) {
		oneClickData = new OneClickRegisterData();
		oneClickData.setPaymentCode(paymentCode);
		oneClickData.setStoreCode(storeCode);
	}

	public void withCreditCard(String cardHolderName, String cardNumber, String cardExpiration) {
		oneClickData.setCardHolderName(cardHolderName);
		oneClickData.setCardNumber(cardNumber);
		oneClickData.setExpirationDate(cardExpiration);
	}

	public void forEmail(String buyerEmail) {
		oneClickData.setBuyerEmail(buyerEmail);
	}

}

File: net/YapayGatewayLib/builder/OneClickBuilder.cs
using System;
using YapayGatewayLib.Model;

namespace YapayGatewayLib.Builder
{
    public class OneClickBuilder
    {

        private static OneClickRegisterData registerData;

        public static OneClickRegisterData build()
        {
            OneClickRegisterData newRegisterData = registerData;
            registerData = null;

            return newRegisterData;
        }


        public static void NewOneClickRegister(string storeCode, long paymentCode)
        {
            registerData = new OneClickRegisterData();
            registerData.storeCode = storeCode;
            registerData.paymentCode = paymentCode;
        }

        public static void WithCreditCard(string cardHoldername, string cardNumber, string cardExpiration)
        {
            registerData.cardHolderName = cardHoldername;
            registerData.cardNumber = cardNumber;
            registerData.expirationDate = cardExpiration;
        }

        public static void ForEmail(string buyerEmail)
        {
            registerData.buyerEmail = buyerEmail;
        }
    }
}